src/main/java/com/volasoftware/tinder/services/implementations/RatingServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.volasoftware.tinder.services.implementations;
+
+import com.volasoftware.tinder.constants.RatingConstant;
+import com.volasoftware.tinder.dtos.FriendDto;
+import com.volasoftware.tinder.dtos.RatingDto;
+import com.volasoftware.tinder.exceptions.RatingNotValidException;
+import com.volasoftware.tinder.models.Rating;
+import com.volasoftware.tinder.repositories.RatingRepository;
+import com.volasoftware.tinder.dtos.RatingResponseDto;
+import com.volasoftware.tinder.services.contracts.AuthenticationService;
+import com.volasoftware.tinder.services.contracts.FriendService;
+import com.volasoftware.tinder.services.contracts.RatingService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+
+@Service
+@RequiredArgsConstructor
+public class RatingServiceImpl implements RatingService {
+
+    private final RatingRepository ratingRepository;
+
+    private final FriendService friendService;
+
+    private final AuthenticationService authenticationService;
+
+    @Override
+    public RatingResponseDto rateFriend(RatingDto ratingDto) {
+        isValidRating(ratingDto);
```

Comment:
This logic is obsolete no need for it. If you handle the validation inside the controller correctly.

Reply:
I agree. This logic and related code will be removed.